src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTableUI.java

```diff
@@ -1884,23 +1884,6 @@ public void paint(Graphics g, JComponent c) {
             comp = comp.getParent();
         }
 
-        if (comp != null && !(comp instanceof JViewport) && !(comp instanceof JScrollPane)) {
-            // We did rMax-1 to paint the same number of rows that are drawn on console
-            // otherwise 1 extra row is printed per page than that are displayed
-            // when there is no scrollPane and we do printing of table
-            // but not when rmax is already pointing to index of last row
-            // and if there is any selected rows
-            if (rMax != (table.getRowCount() - 1) &&
-                    (table.getSelectedRow() == -1)) {
-                // Do not decrement rMax if rMax becomes
-                // less than or equal to rMin
-                // else cells will not be painted
-                if (rMax - rMin > 1) {
-                    rMax = rMax - 1;
-                }
-            }
-        }
-
```

Comment:
Deleting this seems to have regressed JDK-8081491 where 1 extra row is printed per page..
jdk/test/java/awt/print/PageFormat/ImageableAreaTest.java is supposed to print 35 rows which was visible in console but it prints 36 rows..

Reply:
The mentioned test computes 36 rows for frame height of 600 units. The 36th row is 18% visible in console and in printing its 100% visible. I have done the testing for the bug fix with an automated test case LastRowVisible.java and with the below test cases.
1. jdk/test/java/awt/print/PageFormat/ImageableAreaTest.java
2. jdk/test/javax/swing/JTable/LostTextTest.java
3. test/jdk/javax/swing/JTable/TestClearSel.java
The maxRow computations are just fine for console and printable views, the only thing is visibility of the same when setSize of frame is not a multiple of rowHeight of a table......

Reply:
Tested for the below test cases:
1. jdk/test/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java
2. jdk/test/javax/swing/JTable/JTableScrollTest.java
3. Also with SwingSet2 application for verification of JTable console paint and printing.